File: client_app/lib/graphql/device.dart
const String getDeviceQuery = r'''
 query GetDevice($id: String!) {
  getDevice(id: $id) {
    id
    isVerify
    name
    model
    version
    storeId
  }
}
''';

const String newDeviceMutaion = r'''
mutation ($input: DeviceInput) {
  newDevice(input: $input) {
    id
    isVerify
    name
    model
    version
    storeId
  }
}
''';

File: api-gateway/src/schema/resolvers/device.js
export default {
  Query: {
    getDevice: async (root, args, ctx) => {
      try {
        const { data } = await ctx.axios.get(`/devices/` + args.id);
        return data;
      } catch (error) {
        return new Error(error);
      }
    }
  },
  Mutation: {
    newDevice: async (root, args, ctx) => {
      try {
        const { data } = await ctx.axios.post("/devices", { ...args.input });
        return data;
      } catch (error) {
        return new Error(error);
      }
    }
  }
};

// const { data } = await axios.post(`/devices`, { id: "123456" });
